pandas/tools/plotting.py

```diff
@@ -217,10 +217,13 @@ def _maybe_valid_colors(colors):
         # check whether each character can be convertable to colors
         maybe_color_cycle = _maybe_valid_colors(list(colors))
         if maybe_single_color and maybe_color_cycle and len(colors) > 1:
-            msg = ("'{0}' can be parsed as both single color and "
-                   "color cycle. Specify each color using a list "
-                   "like ['{0}'] or {1}")
-            raise ValueError(msg.format(colors, list(colors)))
+            if color[0] == 'C' and len(color) == 2:
```

Comment:
This causes a `ZeroDivisionError` when trying to handle cases where there are more lines than specified colors (or vice versa). [Probably can add a try/except block here](https://github.com/pandas-dev/pandas/blob/da0523a346abd9575ab05746e242ec67c1c442d4/pandas/tools/plotting.py#L239-L244)

In matplotlib, passing `color=''` causes `ValueError: Invalid RGBA argument: ''`. I could add a more meaningful error message for this case as part of this PR.